docs/datafeed.md

```diff
@@ -0,0 +1,136 @@
+
+        bdk.datafeed().subscribe(new RealTimeEventListener() {                                 // (2)
+    
+            @Override
+            public void onMessageSent(V4Initiator initiator, V4MessageSent event) {
+                bdk.messages().send(event.getMessage().getStream(), "<messageML>Hello, World!</messageML>");
+            }
+        });
+
+        bdk.datafeed().start();  
+    }
+}
+```
+
+A more detailed example of the usage of the Datafeed service can be found [here](../symphony-bdk-examples/bdk-core-examples/src/main/java/com/symphony/bdk/examples/DatafeedExampleMain.java).
+
+## Datafeed Configuration
+
+Datafeed Service can be configured by the datafeed field in the configuration file:
+
+```yaml
+datafeed:
+  version: 'v1' # specify datafeed version 'v1' or 'v2'
+  retry:
+    maxAttempts: 6 # maximum number of retry attempts
+    initialIntervalMillis: 2000 # initial interval between two attempts
+    multiplier: 1.5 # interval multiplier after each attempt
+    maxIntervalMillis: 10000 # limit of the interval between two attempts
+```
+
+The minimal configuration for the datafeed service is the version of the datafeed which will be chosen to be use in the bdk.
+If the bot developers want to use the datafeed version 2 in their bot, the version configuration have to be specified as `v2`.
+Otherwise, the datafeed version 1 will be used by default.
+
+Bot developers can also configure a dedicated retry mechanism which will be used only by the datafeed service.
+Basically, the datafeed service retry configuration has the field same as the global retry configuration with the fields for implementing 
+the exponential backoff mechanism.
+
+## Subscribe/Unsubscribe RealTimeEventListener
+
+[RealTimeEventListener](https://javadoc.io/doc/com.symphony.platformsolutions/symphony-bdk-core/latest/com/symphony/bdk/core/service/datafeed/RealTimeEventListener.html) is an interface definition for a callback to be invoked when a real-time event is received from the datafeed.
+This real-time event can be one of these following event types:
+
+- Message Sent
+- Messages Suppressed
+- Symphony Elements Action
+- Shared Wall Post
+- IM/MIM Created
+- Room Created
+- Room Updated Message
+- Room Deactivated Message
+- Room Reactivated Message
+- User Requested to Join Room
+- User Joined Room
+- User Left Room
+- Room Member Promoted To Owner
+- Room Member Demoted From Owner
+- Connection Requested
+- Connection Accepted
+
+The Datafeed Service can subscribe/unsubscribe one or many `RealTimeEventListener` by calling `DatafeedService#subscribe` or
+`DatafeedService#unsubscribe`. For instance:
+
+```java
+public class Example {
+    
+    public static void main(String[] args) {
+        final SymphonyBdk bdk = new SymphonyBdk(loadFromClasspath("/config.yaml"));
+        final RealTimeEventListener listener = new RealTimeEventListener() {
+        @Override
+            public void onMessageSent(V4Initiator initiator, V4MessageSent event) {
```

Comment:
just a thought (out of the scope here), should not this be onMessageReceived? (from the bot point of view)

Reply:
we don't have MessageReceived event, actually, MessageReceived event means a MessageSent with the initiator is someone else not the bot